examples/mujoco/reproduction/trpo/train_trpo.py

```diff
@@ -0,0 +1,189 @@
+    parser.add_argument('--seed', type=int, default=0,
+                        help='Random seed [0, 2 ** 32)')
+    parser.add_argument('--outdir', type=str, default='results',
+                        help='Directory path to save output files.'
+                             ' If it does not exist, it will be created.')
+    parser.add_argument('--steps', type=int, default=2 * 10 ** 6,
+                        help='Total time steps for training.')
+    parser.add_argument('--eval-interval', type=int, default=100000,
+                        help='Interval between evaluation phases in steps.')
+    parser.add_argument('--eval-n-runs', type=int, default=100,
+                        help='Number of episodes ran in an evaluation phase')
+    parser.add_argument('--render', action='store_true', default=False,
+                        help='Render the env')
+    parser.add_argument('--demo', action='store_true', default=False,
+                        help='Run demo episodes, not training')
+    parser.add_argument('--load', type=str, default='',
+                        help='Directory path to load a saved agent data from'
+                             ' if it is a non-empty string.')
+    parser.add_argument('--trpo-update-interval', type=int, default=5000,
+                        help='Interval steps of TRPO iterations.')
+    parser.add_argument('--logger-level', type=int, default=logging.INFO,
+                        help='Level of the root logger.')
+    parser.add_argument('--monitor', action='store_true',
+                        help='Monitor the env by gym.wrappers.Monitor.'
+                             ' Videos and additional log will be saved.')
+    args = parser.parse_args()
+
+    logging.basicConfig(level=args.logger_level)
+
+    # Set random seed
+    chainerrl.misc.set_random_seed(args.seed, gpus=(args.gpu,))
+
+    args.outdir = chainerrl.experiments.prepare_output_dir(args, args.outdir)
+
+    def make_env(test):
+        env = gym.make(args.env)
+        # Use different random seeds for train and test envs
+        env_seed = 2 ** 32 - args.seed if test else args.seed
+        env.seed(env_seed)
+        # Cast observations to float32 because our model uses float32
+        env = chainerrl.wrappers.CastObservationToFloat32(env)
+        if args.monitor:
+            env = gym.wrappers.Monitor(env, args.outdir)
+        if args.render:
+            env = chainerrl.wrappers.Render(env)
+        return env
+
+    env = make_env(test=False)
+    timestep_limit = env.spec.tags.get(
+        'wrapper_config.TimeLimit.max_episode_steps')
+    obs_space = env.observation_space
+    action_space = env.action_space
+    print('Observation space:', obs_space)
+    print('Action space:', action_space)
+
+    assert isinstance(obs_space, gym.spaces.Box)
+
+    # Normalize observations based on their empirical mean and variance
+    obs_normalizer = chainerrl.links.EmpiricalNormalization(
+        obs_space.low.size, clip_threshold=5)
```

Comment:
I didn't find `clip_threshold=5` in the paper, but `5` seems sufficiently large.

Reply:
They used a fork of openai/baselines. TRPO uses `MlpPolicy`, which internally clip normalized observations by 5.
https://github.com/Breakend/baselines/blob/master/baselines/trpo_mpi/run_mujoco.py#L31
https://github.com/Breakend/baselines/blob/master/baselines/ppo1/mlp_policy.py#L25